src/Adapter/Order/QueryHandler/GetOrderForViewingHandler.php

```diff
@@ -0,0 +1,709 @@
+
+        return $order;
+    }
+
+    /**
+     * @param Order $order
+     *
+     * @return OrderCustomerForViewing
+     */
+    private function getOrderCustomer(Order $order): OrderCustomerForViewing
+    {
+        $customer = new Customer($order->id_customer);
+        $gender = new Gender($customer->id_gender);
+        $genderName = '';
+
+        if (Validate::isLoadedObject($gender)) {
+            $genderName = $gender->name[$order->id_lang];
+        }
+
+        $customerStats = $customer->getStats();
+
+        return new OrderCustomerForViewing(
+            $customer->id,
+            $customer->firstname,
+            $customer->lastname,
+            $genderName,
+            $customer->email,
+            new DateTimeImmutable($customer->date_add),
+            Tools::displayPrice(Tools::ps_round(Tools::convertPrice($customerStats['total_orders'], $order->id_currency), PS_ROUND_HALF_UP), (int) $order->id_currency),
+            $customerStats['nb_orders']
+        );
+    }
+
+    /**
+     * @param Order $order
+     *
+     * @return OrderShippingAddressForViewing
+     */
+    public function getOrderShippingAddress(Order $order): OrderShippingAddressForViewing
+    {
+        $address = new Address($order->id_address_delivery);
+        $country = new Country($address->id_country);
+        $stateName = '';
+
+        if ($address->id_state) {
+            $state = new State($address->id_state);
+
+            $stateName = $state->name;
+        }
+
+        return new OrderShippingAddressForViewing(
+            $address->id,
+            $address->firstname,
+            $address->lastname,
+            $address->company,
+            $address->address1,
+            $address->address2,
+            $stateName,
+            $address->city,
+            $country->name[$order->id_lang],
+            $address->postcode,
+            $address->phone,
+            $address->phone_mobile
+        );
+    }
+
+    /**
+     * @param Order $order
+     *
+     * @return OrderInvoiceAddressForViewing
+     */
+    private function getOrderInvoiceAddress(Order $order): OrderInvoiceAddressForViewing
+    {
+        $address = new Address($order->id_address_invoice);
+        $country = new Country($address->id_country);
+        $stateName = '';
+
+        if ($address->id_state) {
+            $state = new State($address->id_state);
+
+            $stateName = $state->name;
+        }
+
+        return new OrderInvoiceAddressForViewing(
+            $address->id,
+            $address->firstname,
+            $address->lastname,
+            $address->company,
+            $address->address1,
+            $address->address2,
+            $stateName,
+            $address->city,
+            $country->name[$order->id_lang],
+            $address->postcode,
+            $address->phone,
+            $address->phone_mobile
+        );
+    }
+
+    /**
+     * @param Order $order
+     *
+     * @return OrderProductsForViewing
```

Comment:
Let's mark this function to be refactored after 😉

Reply:
Why? Do you have anything on your mind? I mean yes, it encapsulates legacy code and this handlers is pretty huge, but why do we need to refactor it? I could be refactored in next major with object model lets say.

Reply:
There's even a `@todo` in the code of this function 😄although we do not modify the legacy magic we encapsulate in handlers, we can at least clean a little bit the code by removing crazy/stupid parts and useless / dead code we can easily remove without making it break.

Even turning `$display_out_of_stock_warning` into camelCase variable is a good thing, although a little one. Let's make this code a little bit cleaner after we've come than we we'v found it.